packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,784 @@
+
+		assert.strictEqual(
+			getIdCompressor(sharedMapContainer2).normalizeToOpSpace(id2),
+			512,
+			"Second container should get second cluster and allocate Id 512",
+		);
+
+		assert.strictEqual(
+			getIdCompressor(sharedMapContainer2).normalizeToOpSpace(id3),
+			513,
+			"Second Id from second container should get second cluster and allocate Id 513",
+		);
+	});
+
+	// IdCompressor is at container runtime level, which means that individual DDSs
+	// in the same container and different DataStores should have the same underlying compressor state
+	it("DDSs in different DataStores have the same compressor state", async () => {
+		const dataStore2 = await factory.createInstance(containerRuntime);
+		mainDataStore.map.set("DataStore2", dataStore2.handle);
+
+		await provider.ensureSynchronized();
+		// 1 Id in the map compressor in the main DataStore, 1 in the cell compressor
+		// in the same DataStore, 1 in the map of DataStore2 should result in 3 local
+		// Ids in the same compressor
+		const compressedIds: SessionSpaceCompressedId[] = [];
+		compressedIds.push(getIdCompressor(sharedMapContainer1).generateCompressedId());
+		compressedIds.push(getIdCompressor(sharedCellContainer1).generateCompressedId());
+		compressedIds.push(getIdCompressor(dataStore2.map).generateCompressedId());
+
+		const decompressedIds: string[] = [];
+		compressedIds.forEach((id) => {
+			const decompressedId = getIdCompressor(sharedMapContainer1).decompress(id);
+
+			// All the compressors point to the same compressor and should all be able to
+			// decompress the local Id to the same decompressed Id
+			[getIdCompressor(sharedCellContainer1), getIdCompressor(dataStore2.map)].forEach(
+				(compressor) => {
+					assert.strictEqual(compressor.decompress(id), decompressedId);
+				},
+			);
+
+			decompressedIds.push(decompressedId);
+		});
+
+		sharedMapContainer1.set("key", "value");
+
+		await provider.ensureSynchronized();
+
+		// Everything should be pointing to the same compressor. All
+		// compressors should have allocated the same number of local Ids.
+		assert.strictEqual(
+			(getIdCompressor(sharedMapContainer1) as any).localIdCount,
+			(getIdCompressor(sharedCellContainer1) as any).localIdCount,
+		);
+
+		assert.strictEqual(
+			(getIdCompressor(sharedMapContainer1) as any).localIdCount,
+			(getIdCompressor(dataStore2.map) as any).localIdCount,
+		);
+
+		decompressedIds.forEach((id, index) => {
+			// All compressors should be able to recompress the decompressed Ids
+			// back to the SessionSpace compressed Id: [-1, -2, -3]
+			const compressedId = getIdCompressor(sharedMapContainer1).recompress(id);
+			assert.strictEqual(compressedIds[index], compressedId);
+
+			[getIdCompressor(sharedCellContainer1), getIdCompressor(dataStore2.map)].forEach(
+				(compressor) => {
+					assert.strictEqual(compressedId, compressor.recompress(id));
+				},
+			);
+		});
+	});
+
+	it("has no compressor if not enabled", async () => {
```

Comment:
Maybe make this the first test? Not a big deal but its kind of in the middle of tests where its enabled :)

Reply:
Yeah I changed the order when I moved the tests out of other files into this one. I put the new ones at the bottom now and the order makes more sense now I think.